src/ucp/proto/proto_init.c

```diff
@@ -123,24 +122,21 @@ ucp_proto_perf_envelope_make(const ucp_proto_perf_list_t *perf_list,
     ucs_status_t status;
     size_t midpoint;
     double x_sample, x_intersect;
-    uint64_t mask;
-
-    ucs_assert_always(perf_list_length < 64);
-    mask = UCS_MASK(perf_list_length);
 
     do {
-        ucs_assert(mask != 0);
-
         /* Find best trend at the 'start' point */
         best.index  = UINT_MAX;
         best.result = DBL_MAX;
-        ucs_for_each_bit(curr.index, mask) {
-            x_sample    = start + UCP_PROTO_MSGLEN_EPSILON;
-            curr.result = ucs_linear_func_apply(perf_list_ptr[curr.index],
+        x_sample    = start;
+        if (x_sample < range_end) {
+            x_sample += UCP_PROTO_MSGLEN_EPSILON;
+        }
+        ucs_for_each_bit(curr.index, funcs_mask) {
+            curr.result = ucs_linear_func_apply(funcs[curr.index],
                                                 x_sample);
             ucs_assert(curr.result != DBL_MAX);
-            if ((best.index == UINT_MAX) ||
-                ((curr.result < best.result) == convex)) {
+            if (!isnan(curr.result) && ((best.index == UINT_MAX) ||
```

Comment:
This handling is required for reconfig proto since it has `f(x) = inf + inf * x` performance. While applying this performance to 0 point we face `inf * 0` which gives `curr.result == NaN`. The current solution isn't perfect since it disallow to choose reconfig for 0 point.

Maybe it worth setting `curr.result=inf` in case if it is NaN or do not add `func.m * x` to result if `x==0` for `ucs_linear_func_apply`.